src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3152,3 +3166,45 @@ void Creature::SetCannotReachTarget(bool cannotReach)
     if (cannotReach)
         sLog->outDebug(LOG_FILTER_UNITS, "Creature::SetCannotReachTarget() called with true. Details: %s", GetDebugInfo().c_str());
 }
+
+bool Creature::CanPeriodicallyCallForAssistance() const
+{
+    // Unable to call for assistance
+    if (HasUnitState(UNIT_STATE_STUNNED | UNIT_STATE_DIED | UNIT_STATE_CONFUSED | UNIT_STATE_POSSESSED))
```

Comment:
I think creatures should still aggro while stunned/confused, needs testing I guess

Reply:
ok

Reply:
Do you know the status code of "change sheep"?

Reply:
I'm pretty sure sheeps aggro, the logic is more like the other creature notices that this creature is attacked/fightning rather than the creature itself calling the others

Reply:
Just asked a few friends that play mage, they confirm that sheeps aggro

Reply:
I mean, creatures should not call stun and sheep changing creatures to fight